skipchain/skipchain.go

```diff
@@ -472,40 +466,33 @@ func (s *Service) lenSkipBlocks() int {
 // saves the actual identity
 func (s *Service) save() {
 	log.Lvl3("Saving service")
-	b, err := network.Marshal(s.SkipBlockMap)
+	err := s.Save("skipblocks", s.SkipBlockMap)
 	if err != nil {
-		log.Error("Couldn't marshal service:", err)
-	} else {
-		err = ioutil.WriteFile(s.path+"/skipchain.bin", b, 0660)
-		if err != nil {
-			log.Error("Couldn't save file:", err)
-		}
+		log.Error("Couldn't save file:", err)
 	}
 }
 
 // Tries to load the configuration and updates the data in the service
 // if it finds a valid config-file.
 func (s *Service) tryLoad() error {
-	configFile := s.path + "/skipchain.bin"
-	b, err := ioutil.ReadFile(configFile)
-	if err != nil && !os.IsNotExist(err) {
-		return fmt.Errorf("Error while reading %s: %s", configFile, err)
+	msg, err := s.Load("skipblocks")
+	if err != nil {
+		return err
 	}
-	if len(b) > 0 {
-		_, msg, err := network.Unmarshal(b)
-		if err != nil {
-			return fmt.Errorf("Couldn't unmarshal: %s", err)
-		}
-		log.Lvl3("Successfully loaded")
-		s.SkipBlockMap = msg.(*SkipBlockMap)
+	if msg == nil {
```

Comment:
I think it should return an error such as https://golang.org/pkg/os/#Open and https://golang.org/pkg/io/ioutil/#ReadFile